OpenSim/Common/Exception.h

```diff
@@ -67,14 +91,29 @@ class OSIMCOMMON_API Exception  : public std::exception {
     std::string _file;
     /** Line number at which the error occurred. */
     int _line;
+    /** Delimiter/separator used to separate different parts of the message.  */
+    static const std::string _msgSeparator;
 
 //=============================================================================
 // METHODS
 //=============================================================================
 public:
     // CONSTRUCTORS
-    Exception(const std::string &aMsg="",const std::string &aFile="",int aLine=-1);
+    Exception(const std::string &aMsg="",
+              const std::string &aFile="",
+              int aLine=-1);
+
+    /** Call this constructor from derived classes to add file, line and 
+    function information to the error message                                 */
+    Exception(const std::string& file,
```

Comment:
It does seem better if the `addMessage()` call could be eliminated, but there is a technical problem that needs to be considered carefully -- it is very important *not* to format the message if the exception is *not* thrown. It is time- and heap-intensive to format a message and you definitely don't want to do that on the off chance some index might go out of range!